modules/cert-manager-acme-dns01-ambient-gcp.adoc

```diff
@@ -0,0 +1,133 @@
++
+This step is required only when the target public-hosted zone overlaps with the cluster's default private-hosted zone.
+
+.. Edit the `CertManager` resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit certmanager cluster
+----
+
+.. Add a `spec.controllerConfig` section with the following override arguments:
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: CertManager
+metadata:
+  name: cluster
+  ...
+spec:
+  ...
+  controllerConfig:                                <1>
+    overrideArgs:
+      - '--dns01-recursive-nameservers-only'       <2>
+----
+<1> Add the `spec.controllerConfig` section.
+<2> Specify to only use recursive nameservers instead of checking the authoritative nameservers associated with that domain.
+
+.. Save the file to apply the changes.
+
+. Optional: Create a namespace for the issuer.
+
+.. Create a YAML file that defines a `Namespace` object:
++
+.Example `namespace.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: my-issuer-namespace <1>
+----
+<1> Specify the namespace for the issuer.
+
+.. Create the `Namespace` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f namespace.yaml
+----
+
+. Create an issuer.
+
+.. Create a YAML file that defines the `Issuer` object:
++
+.Example `issuer.yaml` file
+[source,yaml]
+----
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: issuer-acme-dns01-clouddns <1>
+spec:
+  acme:
+    preferredChain: ""
+    privateKeySecretRef:
+      name: <secret_private_key> <2>
+    server: https://acme-staging-v02.api.letsencrypt.org/directory <3>
+    solvers:
+    - dns01:
+        cloudDNS:
+          project: <project_id> <4>
+----
+<1> Provide a name for the issuer.
+<2> Replace `<secret_private_key>` with the name of the secret to store the ACME account private key in.
+<3> Specify the URL to access the ACME server's `directory` endpoint. This example uses the _Let's Encrypt_ staging environment.
+<4> Replace `<gcp_project_id>` with the name of the project.
+
+.. Create the `Issuer` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f issuer.yaml
+----
+
+. Create a certificate.
+
+.. Create a YAML file that defines the `Certificate` object:
++
+.Example `certificate.yaml` file
+[source,yaml]
+----
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: my-tls-cert <1>
+spec:
+  secretName: my-tls-cert <2>
+  issuerRef:
+    name: issuer-acme-dns01-clouddns <3>
+  dnsNames:
+  - '<domain_name>' <4>
+----
+<1> Provide a name for the certificate.
+<2> Specify the name of the secret to create that will contain the certificate.
+<3> Specify the name of the issuer that you created.
+<4> Replace `<domain_name>` with your domain name.
+
+.. Create the `Certificate` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f certificate.yaml
+----
```

Comment:
Since, we have 5 different cases of users having to create a certificate (all of which still use dns-01) it'd possibly be preferable to keep Certificate manifests explicitly in each step with complete values, otherwise we end up confusing the users in multitude of ways. Also, ambient in GCP and AWS still needs different dns01 challenge fields eg, gcp-project-id in GCP, while in explicit dns-01 it's very different, would be handy to keep different Certificate variants in one page.

Reply:
Certificate manifests for different dns01 challenges **seem not different, seems only a referenced issuer name changes.** It is OK to duplicate certificate manifests with only referenced issuer name changes. But was thinking whether there is better efficient documenting way. So commented above.

Reply:
I misunderstood and thought that we were wanting to change Issuer manifests in ambient. My bad, yeah that is good idea +1 to the original idea, thanks!

Reply:
@xenolinux WDYT about above original review suggestion? Will you update this time?
Or, if you would like to revisit like another similar issue https://github.com/openshift/openshift-docs/pull/59702#discussion_r1242120513 , it is fine, if so, could you create a Jira tracker for them?

Reply:
Hoi @xingxingxia I will create a Jira tracker to revisit the issue.